nanoFirmwareFlasher.Tool/Options.cs

```diff
@@ -341,17 +339,51 @@ public class Options
             HelpText = "Reads details from connected device.")]
         public bool DeviceDetails { get; set; }
 
+        [Option(
+            "showfwonly",
+            Required = false,
+            Default = false,
+            HelpText = "Show which firmware to use for a device without deploying anything.")]
+        public bool ShowFirmwareOnly { get; set; }
+
         [Option(
             "filedeployment",
             Required = false,
             Default = null,
             HelpText = "JSON file containing file deployment settings.")]
         public string FileDeployment { get; set; }
 
+        [Option(
+            "fwarchivepath",
+            Required = false,
+            Default = null,
+            HelpText = "Path of the directory where the firmware is archived.")]
+        public string FwArchivePath { get; set; }
+
+        [Option(
+            "updatefwarchive",
+            Required = false,
+            Default = false,
+            HelpText = "Copy the firmware from the online repository to the firmware archive directory; do not update the firmware on a connected device.")]
+        public bool UpdateFwArchive { get; set; }
+
+        [Option(
+            "fromfwarchive",
+            Required = false,
+            Default = false,
+            HelpText = "Get the firmware from the firmware archive rather than from the online repository.")]
+        public bool FromFwArchive { get; set; }
+
+        [Option(
+            "suppressnanoffversioncheck",
+            Required = false,
+            Default = false,
+            HelpText = $"Do not check whether a new version of {_APPLICATIONALIAS} is available.")]
+        public bool SuppressNanoFFVersionCheck { get; set; }
```

Comment:
Why the need to suppress the version check?
If there is no network connection or similar it won't harm the CLI execution.

Reply:
Yes, it harms the execution of the CLI. It takes up a large fraction of a second and it is completely useless in scenarios where you know you are not going to update the tool. If Laurent is concerned about the performance implications of a lock() statement: this check takes up a lot more time than a lock().

Reply:
I don't think that's an issue, but OK, let's go with it.
Now, the option name takes more time to write than the operation its meant to suppress takes to execute... :D 
Can we find a shorter name without making it sound that the version check is for the tool not the target?